internal/fleet/agent_policy/models.go

```diff
@@ -79,11 +81,20 @@ func (model *agentPolicyModel) populateFromAPI(ctx context.Context, data *kbapi.
 	model.SupportsAgentless = types.BoolPointerValue(data.SupportsAgentless)
 	if data.InactivityTimeout != nil {
 		// Convert seconds to duration string
-		d := time.Duration(*data.InactivityTimeout * float32(time.Second)).Truncate(time.Second)
+		seconds := int64(*data.InactivityTimeout)
+		d := time.Duration(seconds) * time.Second
```

Comment:
This code is pretty confusing, where the duration is created incorrectly (using seconds, rather than nanoseconds, and then corrected. Fix up this case, and the `UnenrollTimeout` case below. 

```suggestion
		seconds := int64(*data.InactivityTimeout) * time.Second
		d := time.Duration(seconds).Truncate(time.Second)
```